plugins/catalog/src/components/EntityContextMenu/EntityContextMenu.tsx

```diff
@@ -136,18 +128,12 @@ export function EntityContextMenu(props: EntityContextMenuProps) {
       >
         <MenuList>
           {extraItems}
-          <MenuItem
-            onClick={() => {
-              onClose();
-              onUnregisterEntity();
-            }}
-            disabled={disableUnregister}
-          >
-            <ListItemIcon>
-              <CancelIcon fontSize="small" />
-            </ListItemIcon>
-            <ListItemText primary="Unregister entity" />
-          </MenuItem>
+          <UnregisterEntity
+            unregisterEntityOptions={UNSTABLE_contextMenuOptions}
+            isUnregisterAllowed={isAllowed}
```

Comment:
Not sure why we pass in this, when we do the logic inside the component with the check for the permissions again. Either remove this prop entirely, or remove the logic from inside the component I think?

Would maybe prefer to remove the prop entirely?